src/firmware/firmware.c

```diff
@@ -162,6 +162,11 @@ k4a_result_t firmware_create(char *device_serial_number, bool resetting_device,
         if (K4A_SUCCEEDED(result))
         {
             result = TRACE_CALL(colormcu_create(container_id, &firmware->colormcu));
+            if ((resetting_device) && K4A_FAILED(result))
```

Comment:
Can we double check that colormcu is not being updated if create fails/it will stay null when the create fails?

Reply:
Yes, it is NULL'd out by colormcu_create in the event of a failure. Good question